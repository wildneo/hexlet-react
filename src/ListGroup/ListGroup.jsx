/* eslint-disable react/prop-types */
/* eslint-disable react/prefer-stateless-function */
import React from 'react';


export default class ListGroup extends React.Component {
  render() {
    const { children } = this.props;

    return (
      <ul className="list-group">
        {React.Children.map(children, (child) => (
          <li className="list-group-item">
            {child}
          </li>
        ))}
      </ul>
    );
  }
}
