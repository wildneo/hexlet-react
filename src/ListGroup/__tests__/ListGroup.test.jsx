import React from 'react';
import renderer from 'react-test-renderer';
import ListGroup from '../ListGroup';

test('ListGroup 1', () => {
  const dom = (
    <ListGroup>
      <p>one</p>
      <p>two</p>
    </ListGroup>
  );

  const component = renderer.create(dom);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('ListGroup 2', () => {
  const dom = (
    <ListGroup>
      <a href="google">google</a>
      <a href="yandex">yandex</a>
      <a href="mail">mail</a>
    </ListGroup>
  );

  const component = renderer.create(dom);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
