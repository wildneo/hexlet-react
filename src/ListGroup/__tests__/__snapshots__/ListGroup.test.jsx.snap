// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGroup 1 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item"
  >
    <p>
      one
    </p>
  </li>
  <li
    className="list-group-item"
  >
    <p>
      two
    </p>
  </li>
</ul>
`;

exports[`ListGroup 2 1`] = `
<ul
  className="list-group"
>
  <li
    className="list-group-item"
  >
    <a
      href="google"
    >
      google
    </a>
  </li>
  <li
    className="list-group-item"
  >
    <a
      href="yandex"
    >
      yandex
    </a>
  </li>
  <li
    className="list-group-item"
  >
    <a
      href="mail"
    >
      mail
    </a>
  </li>
</ul>
`;
